services/fileuploader/fileuploader.go

```diff
@@ -98,6 +104,13 @@ func (p *provider) updateLoop(ctx context.Context, backendConfig backendconfig.B
 				bucket = overrideWithSettings(defaultBucket.Config, settings, workspaceId)
 			} else {
 				bucket = getDefaultBucket(ctx, config.GetString("JOBS_BACKUP_STORAGE_PROVIDER", "S3"))
+				switch c.Settings.DataRetention.RetentionPeriod {
+				case "default":
+					bucket.Config["prefix"] = config.GetString("JOBS_BACKUP_DEFAULT_PREFIX", "7dayretention")
+				case "full":
+					bucket.Config["prefix"] = config.GetString("JOBS_BACKUP_PREFIX", "")
+				default:
```

Comment:
```suggestion
				case "full":
				default:
```
Minor: Since this is a copy of the `default`, we might have to change two places for every change.